### PR TITLE
Multiplatform support 

### PR DESCRIPTION
**Pull request by natsukagami**
- Restructure gears into cross-platform with sbt-crossproject
- Document and include build dependencies for Native
    - Use Boehm GC as Immix is crashing sometimes
- Implement a ForkJoin-executor-based AsyncSupport instance
- Implement Timer and some efforts to make tests run

Some miscellaneous stuff:
- Get other tests to run on all platforms
- Properly check for completion in future addListener

### CODE REVIEW

**Comment by natsukagami** [comment]
TODO: Update CI so it builds dependencies before testing

File: shared/src/main/scala/async/Timer.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * Note that the timer does not start ticking until `start` is called (which is a blocking operation, until the timer is cancelled).
+ * 
+ * You might want to manually `cancel` the timer, so that it gets garbage collected (before the enclosing `Async` scope ends).
+*/
+class Timer(tickDuration: Duration) extends Cancellable {
+  var isCancelled = false
+
+  private class Source extends Async.OriginalSource[this.TimerEvent] {
+    def tick() = synchronized {
+      listeners.foreach(_.completeNow(TimerEvent.Tick, this))
```
Comment (m8nmueller) [inline shared/src/main/scala/async/Timer.scala:26]: This violates the source-attached-listener contract as the listener will be completed multiple times.

File: shared/src/main/scala/async/Timer.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      if isCancelled then k.completeNow(TimerEvent.Cancelled, this)
+      else
+        Timer.this.synchronized:
+          if isCancelled then k.completeNow(TimerEvent.Cancelled, this)
+          else listeners += k
+  }
+  private val _src = Source()
+
+  /** Ticks of the timer are delivered through this source. Note that ticks are ephemeral. */
+  val src: Async.Source[this.TimerEvent] = _src
```
Comment (m8nmueller) [inline shared/src/main/scala/async/Timer.scala:42]: Don't know whether it's actually better (probably a bit) but it feels weird to have to times the same val.
```suggestion
  inline final def src: Async.Source[this.TimerEvent] = _src
```

File: shared/src/main/scala/async/Timer.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  def start()(using Async, AsyncOperations): Unit =
```
Comment (m8nmueller) [inline shared/src/main/scala/async/Timer.scala:44]: Why do you want the timer to be ~blocking~ suspending? Instead, one could use the `Async`s scheduler to run later.

**Pull request merged**